lib/web_ui/lib/src/engine/compositor/fonts.dart

```diff
@@ -14,25 +14,76 @@ const String _robotoUrl =
 
 /// Manages the fonts used in the Skia-based backend.
 class SkiaFontCollection {
-  final List<Future<ByteBuffer>> _loadingFontBuffers = <Future<ByteBuffer>>[];
-  final List<Uint8List> _dynamicallyLoadedFonts = <Uint8List>[];
+  final List<Future<_RegisteredFont>> _unloadedAssetFonts =
+      <Future<_RegisteredFont>>[];
+  final List<_RegisteredFont> _assetFonts = <_RegisteredFont>[];
+  final List<_RegisteredFont> _dynamicallyLoadedFonts = <_RegisteredFont>[];
+
+  /// A mapping from the name a font was registered with, to the family name
+  /// embedded in the font's bytes (the font's "actual" name).
+  ///
+  /// For example, a font may be registered in Flutter assets with the name
+  /// "MaterialIcons", but if you read the family name out of the font's bytes
+  /// it is actually "Material Icons". Skia works with the actual names of the
+  /// fonts, so when we create a Skia Paragraph with Flutter font families, we
+  /// must convert them to their actual family name when we pass them to Skia.
+  final Map<String, String> fontFamilyOverrides = <String, String>{};
 
   final Set<String> registeredFamilies = <String>{};
 
   Future<void> ensureFontsLoaded() async {
+    if (_unloadedAssetFonts.isNotEmpty) {
+      _assetFonts.addAll(
+          (await Future.wait(_unloadedAssetFonts)).where((x) => x != null));
+      _unloadedAssetFonts.clear();
+    }
+    List<_RegisteredFont> registeredFonts = <_RegisteredFont>[
+      ..._assetFonts,
+      ..._dynamicallyLoadedFonts,
+    ];
+
+    fontFamilyOverrides.clear();
+
+    for (_RegisteredFont font in registeredFonts) {
```

Comment:
Suggestion: move into a function called _registerFontFamilyOverrides(). You won't need registerFonts allocation or spread operator code above -> smaller code size.

Reply:
Done